File: cs/src/Connections/SshKeepAliveEventArgs.cs
// <copyright file="SshKeepAliveFailureEventArgs.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license.
// </copyright>

using System;

namespace Microsoft.DevTunnels.Connections;

/// <summary>
/// Event raised when a keep-alive message respose is or is not received.
/// </summary>
public class SshKeepAliveEventArgs : EventArgs
{
    /// <summary>
    /// Create a new instance of <see cref="SshKeepAliveEventArgs"/>.
    /// </summary>
	public SshKeepAliveEventArgs(int count)
	{
		Count = count;
	}

	/// <summary>
	/// The number of keep-alive messages that have been sent with the same state.
	/// </summary>
	public int Count { get; }
}

File: ts/src/connections/sshKeepAliveEventArgs.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

/**
 * Event args raised when an SSH keep-alive succeeds or fails.
 */
export class SshKeepAliveEventArgs {
    /**
     * The number of keep-alive messages that have been sent with the same state.
     */
    public readonly count: number;

    public constructor(count: number) {
        this.count = count;
    }
}
